extensions/positron-proxy/src/positronProxy.ts

```diff
@@ -325,6 +333,22 @@ export class PositronProxy implements Disposable {
 				return;
 			}
 
+			// We don't have an existing proxy server for the target origin, so start a new one.
+			this.startNewProxyServer(contentRewriter).then(({ externalUri, finishProxySetup }) => {
+				finishProxySetup(targetOrigin).then(() => {
```

Comment:
This also needs a catch block unless it's guaranteed `finishProxySetup` can't throw (doesn't throw itself but not sure about the library calls?)